dask/dataframe/io/parquet.py

```diff
@@ -939,15 +941,33 @@ def read_parquet(path, columns=None, filters=None, categories=None, index=None,
     --------
     to_parquet
     """
-    read = get_engine(engine)['read']
-
-    fs, fs_token, paths = get_fs_token_paths(path, mode='rb',
-                                             storage_options=storage_options)
-
-    if isinstance(path, string_types) and len(paths) > 1:
-        # Sort paths naturally if multiple paths resulted from a single
-        # specification (by '*' globbing)
-        paths = sorted(paths, key=natural_sort_key)
+    is_ParquetFile = False
+    try:
+        import fastparquet
+        if isinstance(path, fastparquet.api.ParquetFile):
+            assert (path.file_scheme == 'hive'), "Only 'hive' is currently supported as the ParquetFile's file scheme"
+            is_ParquetFile = True
+    except ImportError:
+            pass
+
+    if is_ParquetFile:
+        read = get_engine('fastparquet')['read']
+        fs, fs_token, paths = get_fs_token_paths(
+                                            path.fn.split('_metadata')[0],
```

Comment:
I don't think this will work for remote files: the filename doesn't (necessarily) include the protocol. Maybe in this line, you could error if it does *not* include a protocol. 
The actual reference kept is `.open`, which is an instance method - you could look at `pf.open.__self__` (probably only for non-local files); using this feels rather hacky.

Reply:
Right, in the default case, `pf.open` is the normal builtin `open` function; but in the general case, the user constructs the instance something like `pf = ParquetFile(path, open_with=s3.open)`, where `s3` is a file-system instance; any method/function returning file-like objects would do here.
The `fs` returned by `get_fs_token_paths` is something different again, like a `DaskS3FileSystem` for s3 or `LocalFileSystem` in your case (no protocol). So the situation is a little complicated! Again, it may be reasonable to only cover the simple case here, of local files with default filesystem.